src/Draco.Compiler/Internal/Symbols/FunctionSymbol.cs

```diff
@@ -74,6 +74,22 @@ internal abstract partial class FunctionSymbol : Symbol, ITypedSymbol
     /// </summary>
     public virtual bool IsVirtual => false;
 
+    public override Api.Semantics.Visibility Visibility
+    {
+        get
+        {
+            var syntax = this.DeclaringSyntax as FunctionDeclarationSyntax;
+            if (syntax is null) return Api.Semantics.Visibility.Internal; // Default
+            return syntax.VisibilityModifier?.Kind switch
+            {
+                null => Api.Semantics.Visibility.Private,
+                TokenKind.KeywordInternal => Api.Semantics.Visibility.Internal,
+                TokenKind.KeywordPublic => Api.Semantics.Visibility.Public,
+                _ => throw new System.InvalidOperationException(nameof(syntax.VisibilityModifier.Kind)),
+            };
```

Comment:
Duplicate code (with VariableSymbol), factor this out into a protected static function inside Symbol or something that can take a nullable kind